web/pandas/pdeps/00xx-string-dtype.md

```diff
@@ -0,0 +1,247 @@
+3. The version that is not backed by PyArrow can reuse the existing numpy
+   object-dtype backed StringArray for its implementation.
+4. We update installation guidelines to clearly encourage users to install
+   pyarrow for the default user experience.
+
+### Default inference of a string dtype
+
+By default, pandas will infer this new string dtype for string data (when
+creating pandas objects, such as in constructors or IO functions).
+
+The existing `future.infer_string` option can be used to opt-in to the future
+default behaviour:
+
+```python
+>>> pd.options.future.infer_string = True
+>>> pd.Series(["a", "b", None])
+0      a
+1      b
+2    NaN
+dtype: string
+```
+
+This option will be expanded to also work when PyArrow is not installed.
+
+### Missing value semantics
+
+Given that all other default data types use NaN semantics for missing values,
+this proposal says that a new default string dtype should still use the same
+default semantics. Further, it should result in default data types when doing
+operations on the string column that result in a boolean or numeric data type
+(e.g., methods like `.str.startswith(..)` or `.str.len(..)`, or comparison
+operators like `==`, should result in default `int64` and `bool` data types).
+
+Because the original `StringDtype` implementations already use `pd.NA` and
+return masked integer and boolean arrays in operations, a new variant of the
+existing dtypes that uses `NaN` and default data types is needed.
+
+### Object-dtype "fallback" implementation
+
+To avoid a hard dependency on PyArrow for pandas 3.0, this PDEP proposes to keep
+a "fallback" option in case PyArrow is not installed. The original `StringDtype`
+backed by a numpy object-dtype array of Python strings can be used for this, and
+only need minor updates to follow the above-mentioned missing value semantics
+([GH-58451](https://github.com/pandas-dev/pandas/pull/58451)).
+
+For pandas 3.0, this is the most realistic option given this implementation is
+already available for a long time. Beyond 3.0, we can still explore further
+improvements such as using nanoarrow or NumPy 2.0, but at that point that is an
+implementation detail that should not have a direct impact on users (except for
+performance).
+
+### Naming
+
+Given the long history of this topic, the naming of the dtypes is a difficult
+topic.
+
+In the first place, we need to acknowledge that most users should not need to
+use storage-specific options. Users are expected to specify `pd.StringDtype()`
+or `"string"`, and that will give them their default string dtype (which
+depends on whether PyArrow is installed or not).
+
+But for testing purposes and advanced use cases that want control over this, we
+need some way to specify this and distinguish them from the other string dtypes.
+Currently, the `StringDtype(storage="pyarrow_numpy")` is used, where
+"pyarrow_numpy" is a rather confusing option.
+
+TODO see if we can come up with a better naming scheme
```

Comment:
`StringDtype(storage="pyarrow", semantics="numpy")`?  or instead of semantics, could use "na_value=np.nan`

Reply:
If i'm understanding correctly about the motivation for the change in dtype (improved overall user experience), then moving forward I suspect that when we can have improved/native dtypes for other data types (nested, date, etc) that the same logic would need to apply, i.e. we would need to have a variants of these with NumPy semantics.

Now this probably falls under PDEP-13 but if we have `semantics` as a argument (that users would see and use) we could still end up with columns using different missing value indicators?

Reply:
> `StringDtype(storage="pyarrow", semantics="numpy")`? or instead of semantics, could use "na_value=np.nan`

or maybe "nullable=[True|False]"

However, at the moment, we distinguish the nullable data types for the other dtypes (int, float, etc) with capitalization and so for consistency could also consider string/String as the dtypes.

Reply:
> `StringDtype(storage="pyarrow", semantics="numpy")`? or instead of semantics, could use "na_value=np.nan`

@jbrockmendel good point that we can also use other keywords than just `storage` to make the distinction

> if we have `semantics` as a argument (that users would see and use) we could still end up with columns using different missing value indicators?

Only if users explicitly specify a non-default value for this, and never by default. This is the same with whatever option we come up with (eg also when using `dtype_backend="pyarrow"` or explicitly asking for one of the masked dtypes with `dtype=Int64` or .. you can end up with a DataFrame with columns with mixed semantics)

> we distinguish the nullable data types for the other dtypes (int, float, etc) with capitalization and so for consistency could also consider string/String as the dtypes.

Yeah, only unfortunately to be consistent with the other dtypes where we use capitalization, it would need to be `"string"` for the new NaN-based dtype, and `"String"` for the "nullable" NA-based variant. And so that doesn't help with backwards compatibility, because `"string"` right now means the nullable dtype. Given that, I would personally not use capitalization here (which also only is a solution for the string alias naming, not for the `StringDtype(..)` API)

---

To keep the sub-discussions manageable, I moved this specific topic out of this inline comment thread, and into it's own issue: https://github.com/pandas-dev/pandas/issues/58613